### PR TITLE
NO-ISSUE: pkg/cli/admin/upgrade/recommend: "no known issues" placeholder, etc.

### PR DESCRIPTION
**Pull request by wking**
Some unrelated cleanups bundled together to address @JianLi-RH 's [feedback][1]:

```console
$ git log --format='%h %s' origin/master..
6607f194 pkg/cli/admin/upgrade/recommend: Fix "no updates to <MAJOR>.<MINOR> available" error message
8dc61654 pkg/cli/admin/upgrade/recommend: Expand "URL" to "Release URL"
a46897a5 pkg/cli/admin/upgrade/recommend: "no known issues" placeholder
```

See the commit messages for more on each pivot.

[1]: https://issues.redhat.com/browse/OTA-1273?focusedId=26029621&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-26029621

### CODE REVIEW

File: pkg/cli/admin/upgrade/recommend/examples/4.12.16-longest-not-recommended.output [hunk 1]
```diff
@@ -24,6 +24,6 @@ And 45 older 4.13 updates you can see with '--show-outdated-releases' or '--vers
 
 Updates to 4.12:
   VERSION     ISSUES
-  4.12.64     
-  4.12.63     
+  4.12.64     no known issues
+  4.12.63     no known issues
```
Comment (wking) [inline pkg/cli/admin/upgrade/recommend/examples/4.12.16-longest-not-recommended.output:28]: Currently the semantics for the previously-empty entries was something like:

> As of this cluster's last poll of its `upstream` update service, there may or may not have been any known risks declared by that update service.  If there were risks, we've successfully evaluated all of them and are confident that they don't match this cluster.

That's obviously too much for a table entry, but I'm slightly concerned that if we say `no known issues` but the admin is aware of a risk affecting that update in general, they'll think "what? _I_ know there's a risk" and forget about the CVO's "am I actually exposed to this one?" PromQL evaluation.

But I'm not so concerned about that that I'm sure `no known issues` is too short.  Just concerned enough to float it here for more feedback.  If you think `no known issues` is likely going to be fine, or folks don't have alternatives they're more confident about soon, I am completely fine going with `no known issues`.  We can always revisit in future pull requests, if we hear about admins getting confused.

**Comment by JianLi-RH** [comment]
The PR resolved all my questions:

```
[jianl@jianl-thinkpadt14gen4 417]$ ./oc adm upgrade recommend
Upstream: https://raw.githubusercontent.com/JianLi-RH/ota/refs/heads/main/OTA-1273.json
Channel: candidate-4.18 (available channels: candidate-4.17, channel-a, stable-4.17)

Updates to 4.18:
  VERSION     ISSUES
  4.18.0-ec.3 no known issues

  Version: 4.18.0-ec.2
  Image: quay.io/openshift-release-dev/ocp-release@sha256:a3233d18821af30610bc16d4a3d0f45a5f6bcfbb50eda139b8b82b824a78559f
  Reason: MultipleReasons
  Message: On clusters on default invoker user, this imaginary bug can happen. https://bug.example.com/a
  
  Too many CI failures on this release, so do not update to it https://openshift-release.apps.ci.l2s4.p1.openshiftapps.com/releasestream/4.10.0-0.nightly/release/4.10.0-0.nightly-2021-11-24-075634

And 1 older 4.18 updates you can see with '--show-outdated-releases' or '--version VERSION'.

Updates to 4.17:
  VERSION     ISSUES
  4.17.3      no known issues
  4.17.2      no known issues
[jianl@jianl-thinkpadt14gen4 417]$ ./oc adm upgrade recommend --version 4.18.0-ec.2
Upstream: https://raw.githubusercontent.com/JianLi-RH/ota/refs/heads/main/OTA-1273.json
Channel: candidate-4.18 (available channels: candidate-4.17, channel-a, stable-4.17)

Update to 4.18.0-ec.2 Recommended=False:
Image: quay.io/openshift-release-dev/ocp-release@sha256:a3233d18821af30610bc16d4a3d0f45a5f6bcfbb50eda139b8b82b824a78559f
Release URL: 
Reason: MultipleReasons
Message: On clusters on default invoker user, this imaginary bug can happen. https://bug.example.com/a
  
  Too many CI failures on this release, so do not update to it https://openshift-release.apps.ci.l2s4.p1.openshiftapps.com/releasestream/4.10.0-0.nightly/release/4.10.0-0.nightly-2021-11-24-075634
[jianl@jianl-thinkpadt14gen4 417]$ ^C
[jianl@jianl-thinkpadt14gen4 417]$ ./oc adm upgrade recommend --version 4.18.0-ec.5
Upstream: https://raw.githubusercontent.com/JianLi-RH/ota/refs/heads/main/OTA-1273.json
Channel: candidate-4.18 (available channels: candidate-4.17, channel-a, stable-4.17)
error: no update to 4.18.0-ec.5 available, so cannot display context for the requested release
[jianl@jianl-thinkpadt14gen4 417]$ 
[jianl@jianl-thinkpadt14gen4 417]$ ./oc adm upgrade recommend --show-outdated-releases
Upstream: https://raw.githubusercontent.com/JianLi-RH/ota/refs/heads/main/OTA-1273.json
Channel: candidate-4.18 (available channels: candidate-4.17, channel-a, stable-4.17)

Updates to 4.18:
  VERSION                            ISSUES
  4.18.0-ec.3                        no known issues
  4.18.0-ec.2                        MultipleReasons
  4.18.0-0.nightly-2024-11-06-051814 no known issues

Updates to 4.17:
  VERSION     ISSUES
  4.17.3      no known issues
  4.17.2      no known issues
[jianl@jianl-thinkpadt14gen4 417]$ ./oc adm upgrade recommend --version 4.17.2
Upstream: https://raw.githubusercontent.com/JianLi-RH/ota/refs/heads/main/OTA-1273.json
Channel: candidate-4.18 (available channels: candidate-4.17, channel-a, stable-4.17)

Update to 4.17.2 has no known issues relevant to this cluster.
Image: quay.io/openshift-release-dev/ocp-release@sha256:e63e11532f9ee550219f076530ce449a5e2ce6f6179e5ccdaf55ce68092e7021
Release URL: https://amd64.ocp.releases.ci.openshift.org/releasestream/4-stable/release/4.17.2
[jianl@jianl-thinkpadt14gen4 417]$ 
```

**Comment by wking** [comment]
Rebased around #1907 with 6607f194 -> c33f996d.  No conflicts in the source code, and the example output is all auto-generated.

File: pkg/cli/admin/upgrade/recommend/examples/4.12.16-longest-not-recommended.output (hunk 1 above)
Comment (hongkailiu) [reply pkg/cli/admin/upgrade/recommend/examples/4.12.16-longest-not-recommended.output:27]: How about "no known issues affecting this cluster", or "no affecting issues"?
My intension is to relief the concern, e.g., `4.12.64` may have risks but none of them affects this cluster.

**Review by hongkailiu — COMMENTED** [review]
Just a question. Otherwise, LGTM.

File: pkg/cli/admin/upgrade/recommend/examples/4.12.16-longest-not-recommended.output (hunk 1 above)
Comment (wking) [reply pkg/cli/admin/upgrade/recommend/examples/4.12.16-longest-not-recommended.output]: I've pushed c33f996d7 -> aaa6d774a to pivot to `no known issues relevant to this cluster` to match the existing:

```console
$ git --no-pager grep 'no known issues' origin/master
origin/master:pkg/cli/admin/upgrade/recommend/recommend.go:                                             fmt.Fprintf(o.Out, "Update to %s has no known issues relevant to this cluster.\nImage: %s\nURL: %s\n", update.Release.Version, update.Release.Image, update.Release.URL)
```

Comment (hongkailiu) [reply pkg/cli/admin/upgrade/recommend/examples/4.12.16-longest-not-recommended.output]: Thanks.

**Comment by wking** [comment]
my command is TechPreview, so [the 4h serial timeout][1] must be unrelated.

/retest-required

[1]: https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_oc/1908/pull-ci-openshift-oc-master-e2e-aws-ovn-serial/1857524013362843648

**Pull request merged**